torch/distributed/_shard/sharded_tensor/_ops/chunk.py

```diff
@@ -0,0 +1,35 @@
+import torch
+from torch.distributed._shard.sharded_tensor import (
+    sharded_op_impl,
+    ShardedTensor,
+)
+
+
+@sharded_op_impl(torch.chunk)
+def sharded_chunk(types, args=(), kwargs=None, pg=None):
+    """
+    Handles ``__torch_function__`` dispatch for the chunk op.
+    """
+    input = args[0]
+    chunks = args[1]
+    dim = kwargs.get("dim")
+    dim = dim if dim else 0
+
+    # Validate types
+    if not isinstance(input, ShardedTensor):
+        raise TypeError("input needs to be a ShardedTensor")
+    st_size = list(input.size())  # type: ignore[arg-type]
+    local_tensor = input.local_tensor()
+    tensor_chunks = torch.chunk(local_tensor, chunks, dim)
+    st_size[dim] //= chunks
+    results = []
+    for tensor in tensor_chunks:
```

Comment:
Because inside transformer code, they concatenate k, v, q into one big matrix. But after chunk, it's actually three separate matrix. What I am thinking here is that:
1. If the chunk dim = sharding dim, the current behavior is to first split local tensor and then build a new st on top of. I guess what you are saying is that we want to first split the original tensor and then shard it again, right?
2. If the chunk dim != sharding dim, Then this logic is mathematically equivalent because the current way: to first split local tensor and then create a st is same as to first split the big tensor and then shard it again.

Is my understanding sound make sense to you?